src/compiler/types.ts

```diff
@@ -5765,6 +5765,7 @@ export interface Symbol {
     /** @internal */ exportSymbol?: Symbol; // Exported symbol associated with this symbol
     /** @internal */ constEnumOnlyModule: boolean | undefined; // True if module contains only const enums or other modules with only const enums
     /** @internal */ isReferenced?: SymbolFlags; // True if the symbol is referenced elsewhere. Keeps track of the meaning of a reference in case a symbol is both a type parameter and parameter.
+    /** @internal */ isInitialized?: boolean; // True if the symbol is ever initialized.

```

Comment:
If this is set in the checker, it probably shouldn't be directly set on the `Symbol` (and should be set on `SymbolLinks` instead). Maybe someone else has some thoughts.

Reply:
Alternatively, can you just use `isAssigned` (defined below) which is set in the binder? You'll need to change some existing code to proactively check for whether the symbol also corresponds to a parameter.

Reply:
Based on perf, this is making the Symbol allocated in the next size class up, which is undesirable. So probably on links.

Reply:
Can't we just reuse `isAssigned`?

Reply:
Probably!

Reply:
Updated. It's totally fine to reuse `isAssigned`.